docs/virtual_machines/liveness_and_readiness_probes.md

```diff
@@ -243,3 +250,107 @@ watchdog -t 2000ms -T 4000ms /dev/watchdog
+
+
+**Important:** If the qemu-guest-agent is not installed **and** enabled inside
+the VM, the probe will fail.  Many images don't enabled the agent by default so
+make sure you either run one that does or enable it. 
+
+Make sure to provide enough delay and failureThreshold for the VM and the agent
+to be online.
+
+In the following example the Fedora image does have qemu-guest-agent available
+by default. Nevertheless, in case qemu-guest-agent is not installed, it will be
+installed and enabled via cloud-init as shown in the example below.  Also,
+cloud-init assigns the proper SELinux context, i.e. virt_qemu_ga_exec_t, to the
+`/tmp/healthy.txt` file.  Otherwise, SELinux will deny the attempts to open the
+`/tmp/healthy.txt` file causing the probe to fail.
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: readiness-probe-vm
+  name: readiness-probe
+spec:
+  running: true 
+  template:
+    metadata:
+      labels:
+        kubevirt.io/domain: readiness-probe
+        kubevirt.io/vm: readiness-probe
+    spec:
+      domain:
+        cpu:
+          cores: 1
+          sockets: 1
+          threads: 1
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          rng: {}
+        resources:
+          requests:
+            memory: 1Gi
+      readinessProbe:
+        exec:
+          command: ["cat", "/tmp/healthy.txt"]
+        failureThreshold: 10
+        initialDelaySeconds: 120
+        periodSeconds: 10
+        # Note that timeoutSeconds value does not have any impact before K8s v1.20.
+        timeoutSeconds: 5
+      terminationGracePeriodSeconds: 180
+      volumes:
+        - containerDisk:
+            image: quay.io/containerdisks/fedora
+          name: containerdisk
+        - cloudInitNoCloud:
+            userData: |
+              #cloud-config
+              chpasswd:
+                expire: false
+              password: password
+              user: fedora
+              packages:
+                qemu-guest-agent
+              runcmd:
+                - ["touch", "/tmp/healthy.txt"]
+                - ["sudo", "chcon", "--type", "virt_qemu_ga_exec_t", "/tmp/healthy.txt"]
+                - ["sudo", "systemctl", "enable", "--now", "qemu-guest-agent"]
+          name: cloudinitdisk
+```
+
+Note that, in the above example if SELinux is not installed in your container
+disk image, the command `chcon` should be removed from the VM manifest shown
+below. Otherwise, the `chcon`  command will fail.
+
+The `.status.ready` field will switch to `true` indicating that probes are being
+successful:
+
+```sh 
+kubectl wait vms/readiness-probe --for=condition=Ready --timeout=5m
+```
+
+Additionally, the following command can be used inside the VM to watch the
+incoming qemu-ga commands:
+
+```sh 
+journalctl --follow 
```

Comment:
Is there a -t you could add to single out just qemu-ga log entries?

Reply:
The following command should only display qemu-ga entries: `journalctl _COMM=qemu-ga --follow`. Changed.